llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -335,10 +335,10 @@ Value *VPTransformState::get(VPValue *Def, bool NeedsScalar) {
   } else {
     // Initialize packing with insertelements to start from undef.
     assert(!VF.isScalable() && "VF is assumed to be non scalable.");
-    Value *Undef = PoisonValue::get(VectorType::get(LastInst->getType(), VF));
+    Value *Undef = PoisonValue::get(toVectorizedTy(LastInst->getType(), VF));
```

Comment:
If you want, you could potentially pull out some of these changes into a NFC patch. For example, toVectorTy -> toVectorizedTy and VectorType::get -> toVectorizedTy, etc. Although if you don't think it's worth it I'm happy as it is.

Reply:
I don't think spilling those out would make sense. Within this patch, all of these changes are needed for the core functionality to work (switching the call does change behavior). If I landed these as an NFC, it makes it less obvious that these are required changes for the struct vectorization.